### PR TITLE
Clean up Examples

### PR DESCRIPTION
**Pull request by christos-h**
## Motivation

Our examples were overlapping in functionality, with us having `x` and `x2` or `x-graphql` to distinguish between functionalities that examples illustrate.

## Solution

All examples except for `counter` now use views. The duplication is gone.

### CODE REVIEW

File: examples/counter/Cargo.toml [hunk 1]
```diff
@@ -5,22 +5,26 @@ authors = ["Linera <contact@linera.io>"]
...
 
 [[bin]]
-name = "counter_contract"
+name = "counter_graphql_contract"
 path = "src/contract.rs"
 
 [[bin]]
-name = "counter_service"
-path = "src/service.rs"
+name = "counter_graphql_service"
```
Comment (afck) [inline examples/counter/Cargo.toml:29]: Do we still want the `graphql` in the binary names?

File: examples/counter/src/service.rs [hunk 2]
```diff
@@ -25,11 +26,44 @@ impl Service for Counter {
...
         }
     }
 }
 
+struct MutationRoot;
+
+#[Object]
+impl MutationRoot {
+    #[allow(unused)]
+    async fn execute_operation(&self, operation: u64) -> Vec<u8> {
```
Comment (afck) [inline examples/counter/src/service.rs:50]: Should we call this `increment`?

File: examples/counter/web-frontend/src/App.js [hunk 3]
```diff
@@ -21,7 +21,11 @@ const INCREMENT_COUNTER = gql`
 
 const NOTIFICATION_SUBSCRIPTION = gql`
   subscription {
-    notifications()
+    notifications(
+      chainIds: [
+        "7817752ff06b8266d77df8febf5c4b524cec096bd83dc54f989074fb94f833737ae984f32be2cee1dfab766fe2d0c726503c4d97117eb59023e9cc65a8ecd1f7"
```
Comment (afck) [inline examples/counter/web-frontend/src/App.js:26]: Is the chain ID still necessary?

File: examples/fungible/src/state.rs [hunk 4]
```diff
@@ -3,50 +3,67 @@
...
+            .get(account)
+            .await
+            .expect("Failure in the retrieval")
+            .unwrap_or_default()
     }
 
     /// Credits an `account` with the provided `amount`.
-    pub(crate) fn credit(&mut self, account: AccountOwner, amount: Amount) {
+    pub(crate) async fn credit(&mut self, account: AccountOwner, amount: Amount) {
+        let mut value = self.balance(&account).await;
```
Comment (afck) [inline examples/fungible/src/state.rs:46]: For consistency with `debit`, let's call it `balance` here, too.

File: examples/reentrant-counter/src/contract.rs [hunk 5]
```diff
@@ -69,9 +71,11 @@ impl Contract for ReentrantCounter {
         _forwarded_sessions: Vec<SessionId>,
     ) -> Result<ApplicationCallResult, Self::Error> {
         let increment: u128 = bcs::from_bytes(argument)?;
-        self.value += increment;
+        let mut value = *self.value.get();
+        value += increment;
+        self.value.set(value);
```
Comment (afck) [inline examples/reentrant-counter/src/contract.rs:76]: Could we use `get_mut` here, too?

File: linera-core/src/unit_tests/wasm_worker_tests.rs [hunk 6]
```diff
@@ -126,10 +110,7 @@ where
     .await;
 
     // Load some bytecode.
-    let name_counter = match storage_kind {
-        StorageKind::Simple => "counter",
-        StorageKind::View => "counter2",
-    };
+    let name_counter = "counter";
```
Comment (afck) [inline linera-core/src/unit_tests/wasm_worker_tests.rs:113]: I'd inline that now.

**Review by afck — APPROVED** [review]

File: examples/counter/src/service.rs (hunk 2 above)
Comment (christos-h) [reply examples/counter/src/service.rs:50]: Yes, I'll do this in a new PR since this one is already quite big. #696

**Pull request merged**